tests/Unit/Adapter/Image/ImageValidatorTest.php

```diff
@@ -0,0 +1,118 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Image;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Image\ImageValidator;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageFileNotFoundException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageConstraintException;
+use Tests\Resources\DummyFileUploader;
+
+class ImageValidatorTest extends TestCase
+{
+    /**
+     * @var ImageValidator
+     */
+    private $imageValidator;
+
+    public function setUp()
+    {
+        require_once __DIR__ . '/../../bootstrap.php';
+        $this->imageValidator = new ImageValidator();
+    }
+
+    /**
+     * @dataProvider getInvalidMaxUploadSizesForFile
+     *
+     * @param string $filePath
+     * @param int $maxUploadSize
+     */
+    public function testItThrowsExceptionWhenFileSizeIsLargerThanMaxUploadSize(string $filePath, int $maxUploadSize)
+    {
+        $imageValidator = new ImageValidator($maxUploadSize);
+        $this->expectException(UploadedImageConstraintException::class);
+        $this->expectExceptionCode(UploadedImageConstraintException::EXCEEDED_SIZE);
+
+        $imageValidator->assertFileUploadLimits($filePath);
+    }
+
+    /**
+     * @dataProvider getUnsupportedImageTypes
+     *
+     * @param string $filePath
+     * @param array|null $allowedTypes
+     *
+     * @throws UploadedImageConstraintException
+     * @throws ImageUploadException
+     */
+    public function testItThrowsExceptionWhenUnsupportedImageTypeIsProvided(string $filePath, ?array $allowedTypes): void
+    {
+        $this->expectException(UploadedImageConstraintException::class);
+        $this->expectExceptionCode(UploadedImageConstraintException::UNRECOGNIZED_FORMAT);
+
+        $this->imageValidator->assertIsValidImageType($filePath, $allowedTypes);
+    }
+
+    /**
+     * @dataProvider getInvalidPathsToAFile
+     *
+     * @param string $filePath
+     */
+    public function testItThrowsExceptionWhenFileDoesNotExistByProvidedPath(string $filePath): void
+    {
+        $this->expectException(ImageFileNotFoundException::class);
+        $this->imageValidator->assertIsValidImageType($filePath);
+    }
+
+    public function getInvalidMaxUploadSizesForFile(): Generator
+    {
+        $logoPath = DummyFileUploader::getDummyFilesPath() . 'logo.jpg';
+        $appIconPath = DummyFileUploader::getDummyFilesPath() . 'app_icon.png';
+
+        yield [$logoPath, 2500];
```

Comment:
I'm always impressed with these yield/generator things 😅  I need to start using them to get more used to it

Reply:
Good news I made my first Test class using `yield` https://github.com/PrestaShop/PrestaShop/blob/2638bc413338d9d6cfbd832014c62d108ccc7459/tests/Unit/Adapter/Cart/Comparator/CartProductsComparatorTest.php#L60
😅